boofuzz/primitives/bytes.py

```diff
@@ -0,0 +1,248 @@
+                offset = len(self._fuzz_library) + len(self.this_library) + len(self._magic_debug_values)
+                testcase_nr = self._mutant_index - offset
+                testcases_2a = len(self._fuzz_strings_1byte) * max(0, len(self._original_value) - 0)
+                testcases_2b = len(self._fuzz_strings_2byte) * max(0, len(self._original_value) - 1)
+                testcases_2c = len(self._fuzz_strings_4byte) * max(0, len(self._original_value) - 3)
+                if testcase_nr < testcases_2a:
+                    j = testcase_nr % len(self._fuzz_strings_1byte)
+                    i = testcase_nr // len(self._fuzz_strings_1byte)
+                    self._value = self._original_value[:i] + self._fuzz_strings_1byte[j] + self._original_value[i+1:]
+                elif testcase_nr < testcases_2a + testcases_2b:
+                    testcase_nr -= testcases_2a
+                    j = testcase_nr % len(self._fuzz_strings_2byte)
+                    i = testcase_nr // len(self._fuzz_strings_2byte)
+                    self._value = self._original_value[:i] + self._fuzz_strings_2byte[j] + self._original_value[i+2:]
+                elif testcase_nr < testcases_2a + testcases_2b + testcases_2c:
+                    testcase_nr -= testcases_2a
+                    testcase_nr -= testcases_2b
+                    j = testcase_nr % len(self._fuzz_strings_4byte)
+                    i = testcase_nr // len(self._fuzz_strings_4byte)
+                    self._value = self._original_value[:i] + self._fuzz_strings_4byte[j] + self._original_value[i+4:]
+                else:
+                    # should not be reachable!
+                    assert(False)
+
+            # increment the mutation count.
+            self._mutant_index += 1
+
+            # if the size parameter is disabled, done.
+            if self.size == -1:
+                return True
+
+            # ignore library items greater then user-supplied length.
+            # TODO: might want to make this smarter.
+            if len(self._value) > self.size:
+                continue
+            else:
+                return True
+
+    def num_mutations(self):
+        """
+        Calculate and return the total number of mutations for this individual primitive.
+
+        @rtype:  int
+        @return: Number of mutated forms this primitive can take
+        """
+        num = len(self._fuzz_library) + len(self.this_library) + len(self._magic_debug_values)
+        num += len(self._fuzz_strings_1byte) * max(0, len(self._original_value) - 0)
+        num += len(self._fuzz_strings_2byte) * max(0, len(self._original_value) - 1)
+        num += len(self._fuzz_strings_4byte) * max(0, len(self._original_value) - 3)
+        return num
+
+    def _render(self, value):
+        """
+        Render string value, properly padded.
+        """
+
+        if isinstance(value, six.text_type):
+            value = helpers.str_to_bytes(value)
+
```

Comment:
It seems the condition will be checked again in `helpers.str_to_bytes()`. Maybe using `value = helpers.str_to_bytes(value)` without the condition directly.